internal/app/admin/http/handler_test.go

```diff
@@ -361,6 +387,301 @@ func TestAdminServer_CacheDumpHandler_EntireCacheV3(t *testing.T) {
 	}
 }
 
+func TestAdminServer_CacheDumpHandler_WildcardSuffix(t *testing.T) {
+	for _, url := range []string{"/cache/t*", "/cache/tes*", "/cache/test*"} {
+		ctx := context.Background()
+		mapper := mapper.NewMock(t)
+		upstreamResponseChannelLDS := make(chan *v2.DiscoveryResponse)
+		upstreamResponseChannelCDS := make(chan *v2.DiscoveryResponse)
+		mockScope := tally.NewTestScope("mock_orchestrator", make(map[string]string))
+		client := upstream.NewMock(
+			ctx,
+			upstream.CallOptions{Timeout: time.Second},
+			nil,
+			upstreamResponseChannelLDS,
+			nil,
+			nil,
+			upstreamResponseChannelCDS,
+			func(m interface{}) error { return nil },
+		)
+		orchestrator := orchestrator.NewMock(t, mapper, client, mockScope)
+		assert.NotNil(t, orchestrator)
+
+		req1Node := corev2.Node{
+			Id:      "test-1",
+			Cluster: "test-prod",
+		}
+		gcpReq1 := gcp.Request{
+			TypeUrl: resourcev2.ListenerType,
+			Node:    &req1Node,
+		}
+		ldsRespChannel, cancelLDSWatch := orchestrator.CreateWatch(transport.NewRequestV2(&gcpReq1))
+		assert.NotNil(t, ldsRespChannel)
+
+		req2Node := corev2.Node{
+			Id:      "test-2",
+			Cluster: "test-prod",
+		}
+		gcpReq2 := gcp.Request{
+			TypeUrl: resourcev2.ClusterType,
+			Node:    &req2Node,
+		}
+		cdsRespChannel, cancelCDSWatch := orchestrator.CreateWatch(transport.NewRequestV2(&gcpReq2))
+		assert.NotNil(t, cdsRespChannel)
+
+		listener := &v2.Listener{
+			Name: "lds resource",
+		}
+		listenerAny, err := ptypes.MarshalAny(listener)
+		assert.NoError(t, err)
+		resp := v2.DiscoveryResponse{
+			VersionInfo: "1",
+			TypeUrl:     resourcev2.ListenerType,
+			Resources: []*any.Any{
+				listenerAny,
+			},
+		}
+		upstreamResponseChannelLDS <- &resp
+		gotResponse := <-ldsRespChannel.GetChannel().V2
+		gotDiscoveryResponse, err := gotResponse.GetDiscoveryResponse()
+		assert.NoError(t, err)
+		assert.Equal(t, resp, *gotDiscoveryResponse)
+
+		cluster := &v2.Cluster{
+			Name: "cds resource",
+		}
+		clusterAny, err := ptypes.MarshalAny(cluster)
+		assert.NoError(t, err)
+		resp = v2.DiscoveryResponse{
+			VersionInfo: "2",
+			TypeUrl:     resourcev2.ClusterType,
+			Resources: []*any.Any{
+				clusterAny,
+			},
+		}
+		upstreamResponseChannelCDS <- &resp
+		gotResponse = <-cdsRespChannel.GetChannel().V2
+		gotDiscoveryResponse, err = gotResponse.GetDiscoveryResponse()
+		assert.NoError(t, err)
+		assert.Equal(t, resp, *gotDiscoveryResponse)
+
+		req, err := http.NewRequest("GET", url, nil)
+		assert.NoError(t, err)
+
+		rr := httptest.NewRecorder()
+		handler := cacheDumpHandler(&orchestrator)
+
+		handler.ServeHTTP(rr, req)
+		assert.Equal(t, http.StatusOK, rr.Code)
+
+		body := dateRegex.ReplaceAllString(rr.Body.String(), "\"\"")
+		filecontentsCds, err := ioutil.ReadFile("testdata/entire_cachev2_cds.json")
+		assert.NoError(t, err)
+		filecontentsLds, err := ioutil.ReadFile("testdata/entire_cachev2_lds.json")
+		assert.NoError(t, err)
+		assert.Contains(t, body, string(filecontentsCds))
```

Comment:
this will only match the a single entry, right? Can you add a test case where we match multiple keys and we assert the contents of body exactly?

Reply:
Sorry, can you clarify what you mean here? This matches the generated keys `test_cds` and `test_lds` because they both begin with "t", "tes", and "test".

Reply:
indeed, thanks for clarifying this. Disregard my comment then. This looks good.